kore/src/Kore/Exec/GraphTraversal.hs

```diff
@@ -263,8 +267,8 @@ graphTraversal
             Seq (TState instr config) ->
             Simplifier (StepResult (TState instr config))
         step a q = do
-            next <- transit a
-            if (isStuck next || isFinal next || isStopped next || shouldStop next)
```

Comment:
The problem was that `shouldStop` was used here, but not in `checkLeftUnproven` below where "unproven" (stopped) states are turned into a different result.
A cleaner solution is to turn `Branch` into `Stopped`. The open PR #3192 will refactor this to keep the next states in `Stopped`, too, so no information is lost.